ft: signin with email and password

sostene12 opened the pull request:
#### What does this PR do?
- implement Sign in with Email & Password
#### Description of Task to be completed
- implement sign in with email and password
- implement endpoint documentation
- implement error return during sign in
#### How should this be manually tested?
- clone the repo using git clone repo's URL
- change branch git checkout ft-signin
- create the roles using the roles endpoints like `buyer, vendor or admin`
- create permissions reserved for buyer or the vendor using the permission endpoints like `buyer views-product` or `vendor sella-product`
- create the user like the buyer 
- test signin 
- create a .env file and put the required environment variables from .env.example file
- run in the terminal `npm run db:seeds`,
- make a post request using the provides credentials in swagger docs
#### What are the relevant pivotal tracker stories?
-S ign in with Email & Password cardnbr #5

Luciefifi committed:
ft(roles and permissions):
-add role, permission model
-add junction relation
-make migrations
-make seeders
-add controllers
-add routes
-modify index.js
-modify app.js
-finishes[#12]

sostene12 committed:
ft(roles-permission): roles and permissions

- implement roles and permission relationship
- implement create role and delete role
- implement create and delete permission
- implement assignrole and permission to the user

[WIP ft-roles-permission ]

wakaflorien reviewed (commented):
- i don't get why we need two config files envConfig and config

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

sostene12 committed:
merging with develop

sostene12 committed:
ft(roles-permission): roles-permission

-implement internationalization
-implement swgger doc of the endpoint
-implement enpoints test

[finished ft-roles-permissions ]

sostene12 committed:
ft-roles-permissions
- implement relationships

sostene12 committed:
emails

sostene12 committed:
scripts of test

oliviertech212 reviewed (commented):
please provide clear description  to test this  ft-signin

sostene12 committed:
rebasing

sostene12 committed:
solved  the conflicts
